### PR TITLE
test: read receipts

### PR DESCRIPTION
**Pull request by heitortanoue**
## Proposed changes (including videos or screenshots)

Create a new file dedicated to the read-receipts feature tests and improve its coverage (including some edge-cases introduced in the last major release).

## Further comments

TC-502

### CODE REVIEW

File: apps/meteor/tests/end-to-end/api/24-methods.js [hunk 1]
```diff
@@ -123,6 +126,276 @@ describe('Meteor.methods', function () {
 		});
 	});
 
+	describe('[@getReadReceipts]', () => {
+		it('should fail if not logged in', async () => {
```
Comment (pierre-lehnen-rc) [inline apps/meteor/tests/end-to-end/api/24-methods.js:130]: This request will fail even if the user is logged in, since no params are being passed.
You should send an otherwise valid request and also check the error type to ensure it's the user state that is causing the request to fail.

File: apps/meteor/tests/end-to-end/api/24-methods.js [hunk 2]
```diff
@@ -123,6 +126,276 @@ describe('Meteor.methods', function () {
...
+							id: 'id',
+							msg: 'method',
+						}),
+					})
+					.expect('Content-Type', 'application/json')
+					.expect(200)
+					.expect((res) => {
+						expect(res.body).to.have.a.property('success', true);
+						expect(res.body).to.have.a.property('message').that.is.a('string');
+						console.log('res.body', res.body);
```
Comment (pierre-lehnen-rc) [inline apps/meteor/tests/end-to-end/api/24-methods.js:215]:
```suggestion
```

File: apps/meteor/tests/end-to-end/api/24-methods.js [hunk 3]
```diff
@@ -123,6 +126,276 @@ describe('Meteor.methods', function () {
...
+						expect(res.body).to.have.a.property('message').that.is.a('string');
+
+						const data = JSON.parse(res.body.message);
+						expect(data).to.have.a.property('result').that.is.an('array');
+						expect(data.result.length).to.equal(1);
+						expect(data.result[0]).to.have.property('userId', credentials['X-User-Id']);
+					});
+			});
+
+			it("should read all main room's messages with the invited user", async () => {
```
Comment (pierre-lehnen-rc) [inline apps/meteor/tests/end-to-end/api/24-methods.js:250]: This isn't a test, but a requirement for the next few tests; 
Move them all to a new `describe` section that runs independently from this one, with it's own `before` sub-section.

File: apps/meteor/tests/end-to-end/api/24-methods.js [hunk 4]
```diff
@@ -123,6 +126,276 @@ describe('Meteor.methods', function () {
...
+						expect(res.body).to.have.a.property('message').that.is.a('string');
+
+						const data = JSON.parse(res.body.message);
+						expect(data).to.have.a.property('result').that.is.an('array');
+						expect(data.result.length).to.equal(1);
+						expect(data.result[0]).to.have.property('userId', credentials['X-User-Id']);
+					});
+			});
+
+			it('should read thread messages with the invited user', async () => {
```
Comment (pierre-lehnen-rc) [inline apps/meteor/tests/end-to-end/api/24-methods.js:322]: Same as above, this one is not a test, but data manipulation.

File: apps/meteor/tests/end-to-end/api/24-methods.js [hunk 5]
```diff
@@ -123,6 +126,276 @@ describe('Meteor.methods', function () {
...
+						const data = JSON.parse(res.body.message);
+						expect(data).to.have.a.property('result').that.is.an('array');
+						expect(data.result.length).to.equal(2);
+
+						const receiptsUserIds = [data.result[0].userId, data.result[1].userId];
+						expect(receiptsUserIds).to.have.members([credentials['X-User-Id'], user._id]);
+					});
+			});
+
+			it('should mark the thread as read', async () => {
```
Comment (pierre-lehnen-rc) [inline apps/meteor/tests/end-to-end/api/24-methods.js:377]: This one is also data manipulation, that is not even being validated by anything.

**Review by pierre-lehnen-rc — CHANGES_REQUESTED** [review]
You'll need multiple `describe` sections for all situations you're testing, each one with their own messages in the correct state.

1 - simple message and thread that nobody has read yet
2 - simple message and thread where the room message was read by another user but the thread message was not
3 - another test where everything was read by both users
4 - another test that checks the result of `getReadReceipts` after the call to `readThreads`

**Review by pierre-lehnen-rc — CHANGES_REQUESTED** [review]
Avoid calling `expect` inside the `before` section of tests; if the call fails the test will fail anyway.

**Review by pierre-lehnen-rc — APPROVED** [review]
